src/libs/actions/data-lake.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * A variable parameter for a Cockpit action
+ * @param { string } id - The id of the variable
+ * @param { string } name - The name of the variable
+ * @param { 'string' | 'number' | 'boolean' } type - The type of the variable (string, number or boolean)
+ * @param { string } description - What the variable does or means
+ */
+class CockpitActionParameter {
+  id: string
+  name: string
+  type: 'string' | 'number' | 'boolean'
+  description?: string
+  // eslint-disable-next-line jsdoc/require-jsdoc
+  constructor(id: string, name: string, type: 'string' | 'number' | 'boolean', description?: string) {
+    this.id = id
+    this.name = name
+    this.type = type
+    this.description = description
+  }
+}
+
+const cockpitActionParametersInfo: Record<string, CockpitActionParameter> = {}
+export const cockpitActionParametersData: Record<string, string | number | boolean> = {}
+const cockpitActionParametersListeners: Record<string, ((value: string | number | boolean) => void)[]> = {}
+
+export const getCockpitActionParametersInfo = (id: string): CockpitActionParameter | undefined => {
+  return cockpitActionParametersInfo[id]
+}
+
+export const getAllCockpitActionParametersInfo = (): Record<string, CockpitActionParameter> => {
+  return cockpitActionParametersInfo
+}
+
+export const getCockpitActionParameterInfo = (id: string): CockpitActionParameter | undefined => {
+  return cockpitActionParametersInfo[id]
+}
```

Comment:
Aren't these two functions (getCockpitActionParametersInfo and getCockpitActionParameterInfo) duplicated?

Reply:
They definitely are 😆